File: Hotel/Data/Entities/Room.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace Hotel.Entities
{
    public class Room
    {
        public int RoomId { get; set; }
        public string Name { get; set; }
        public string Floor { get; set; }
        [JsonIgnore] public virtual Category RoomType { get; set; }
        public int RoomTypeId { get; set; }

        public int NumberOfSeats { get; set; }
        public bool HasMiniBar { get; set; }
        public bool IsDeleted { get; set; }
        [NotMapped] public bool IsFree { get; } //ToDo
    }
}

File: Hotel/ClientApp/src/app/_models/room.ts
import { jsonIgnoreReplacer, jsonIgnore } from 'json-ignore';

export class Room {

  @jsonIgnore() public TheNoumber: number; // для нумерации списка комнат(так проще вроде бы)
  @jsonIgnore() public Cost: number;

    constructor(
        public roomId?: number,
        public name?: string,
        public floor?: string,
        public roomTypeId?: number,
        public numberOfSeats?: number,
        public hasMiniBar?: boolean,
        public isFree?: boolean,
        public isDeleted ?: boolean) { }

}
